HomeWork 1.0

davydov-den opened the pull request:
Домашняя работа номер 1. Дата выдачи 19.03.2014
Выполнили Давыдов Денис и Смыков Василий.

davydov-den committed:
HomeWork 1.0

Домашняя работа номер 1. Дата выдачи 19.03.2014
Выполнили Давыдов Денис и Смыков Василий.

davydov-den committed:
Delete README.md

aleksandra-jashina commented on Ceramic Cup:
```diff
@@ -0,0 +1,97 @@
+•	Можно ли за нее взяться.
+
+  Интеграционное тестирование.
+Совместим два модуля : емкость и ручка.
```
Здесь стоит проверить, что кружка вообще устойчиво стоит в нужном положении (т.к. уже добавили ручку, которая может повлиять на баланс)

davydov-den replied:
Согласен, добавил!

skyline-gleb commented:
Пара замечаний:
1) Файл с решением не в папке с именем 1903*
2) У файла должно бы быть расширение

aleksandra-jashina commented on Ceramic Cup:
```diff
@@ -0,0 +1,97 @@
+
+
+Системное тестирование:
+	Отказоустойчивость:
```
Сначала все же стоит провести функциональное тестирование, а затем уже рассматривать менее критичные аспекты.
Не хватает самого функционального тестирования как отдельного этапа: мы проверили выше, что в кружку можно налить жидкость, но еще не знаем, можно ли из нее пить, удержит ли она равновеси в процессе наливания жидкости и т.п.

davydov-den replied:
Добавил!

aleksandra-jashina commented on Ceramic Cup:
```diff
@@ -0,0 +1,97 @@
+    облезет краска и т.д.).
+Локализация: 
+•	Если в комплекте к кружке есть инструкция, то она должна быть переведена на язык того места, где кружка используется;
+•	Проверить можно ли взять за ручку в толстой перчатке (вдруг ее будут использовать в холодных условиях.
```
эЭто все-таки не локализация. Это может быть функциональное требование (если явно оговорено), либо удобство использования (если рассматривать как дополнительный плюс)

davydov-den replied:
Согласен перенес в usability

aleksandra-jashina commented on Ceramic Cup:
```diff
@@ -0,0 +1,97 @@
+•	Проверить кружку, уронив с высоты чуть больше половины человеческого роста (на которой обычно носят кружку);
+•	Проверить не протекает ли кружка (вдруг нальешь горячую жидкость, а она будет капать на пользователя).
+Производительность:
+•	Проверить совпадает ли объем кружки с заявленным;
```
Почему это производительность?

davydov-den replied:
Ну да,стоит убрать.

aleksandra-jashina commented on Ceramic Cup:
```diff
@@ -0,0 +1,97 @@
+•	Проверить не протекает ли кружка (вдруг нальешь горячую жидкость, а она будет капать на пользователя).
+Производительность:
+•	Проверить совпадает ли объем кружки с заявленным;
+•	Проверить насколько быстро можно наливать, выливать жидкость.
```
Что еще может происходить с различной скоростью с жидкостью и кружкой?

davydov-den replied:
Ну а вдруг слишком маленькая горловина у кружки? Что жидкость очень медленно может затекать?

aleksandra-jashina commented on Ceramic Cup:
```diff
@@ -0,0 +1,97 @@
+Проверить ударостойкость кружки.
+
+Time – любые отношения между продуктом и временем.
+Время никаким образом не должно влиять ни на функционал, ни на форму, ни на внешний вид кружки.
```
Можно проверить, окрашиваются ли стенки со временем, стирается ли рисунок

davydov-den replied:
Согласен,добавил!

aleksandra-jashina commented on Ceramic Cup:
```diff
@@ -0,0 +1,97 @@
+•	Вес (кружка не должна иметь слишком большой вес);
+•	Удобство ручки (может ли обычный человек удерживать кружку в воздухе не испытывая лишнего дискомфорта);
+•	Удобство наливания (не мешает ли ручка наливать жидкость, не слишком ли маленькое отверстие в кружке);
+Безопасность:
```
Можно добавить тесты с точки зрения безопасности для пользователя: теплопроводность, форма края кружки

davydov-den replied:
Согласен, добавил!

davydov-den committed:
Исправление ошибок.

Добавлены:
1.  тесты на устойчивость после сборки.
2. функциональные тесты
3. тесты на безпасность
4. тесты для time

Убран один тест на производительность.

Один тест перенесен из локализации в usability.

davydov-den committed:
Rename Ceramic Cup to Ceramic Cup.txt
